source/common/upstream/cluster_manager_impl.cc

```diff
@@ -283,6 +283,11 @@ ClusterManagerImpl::ClusterManagerImpl(
 
   ads_mux_->start();
 
+  // Pause RDS to ensure that we don't send any requests until we've
+  // subscribed to all the RDS resources. The subscriptions happen in the init callbacks,
+  // so we pause RDS until we've completed all the callbacks.
+  ads_mux_->pause(Config::TypeUrl::get().RouteConfiguration);
```

Comment:
The context of previous `pause`: #3399 
It makes me think we can pause it as early as here

Reply:
Is this related to this PR? If it can be a distinct PR, all the better.